### ISSUE TITLE
one or more tests in twisted.trial._dist.test.test_worker corrupt test.log

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10321]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#10321|
|Type|defect|
|Created|2022-03-28 13:19:42Z|

There are 23 tests in `twisted.trial._dist.test.test_worker`:

```
 python -m twisted.trial twisted.trial._dist.test.test_worker
twisted.trial._dist.test.test_worker
  LocalWorkerAMPTests
    test_runError ...                                                      [OK]
    test_runErrorWithFrames ...                                            [OK]
    test_runExpectedFailure ...                                            [OK]
    test_runFailure ...                                                    [OK]
    test_runSkip ...                                                       [OK]
    test_runSuccess ...                                                    [OK]
    test_runUnexpectedSuccesses ...                                        [OK]
    test_stopAfterRun ...                                                  [OK]
    test_testWrite ...                                                     [OK]
  LocalWorkerTests
    test_addresses ...                                                     [OK]
    test_childDataReceived ...                                             [OK]
    test_connectionLost ...                                                [OK]
    test_errReceived ...                                                   [OK]
    test_loseConnection ...                                                [OK]
    test_outReceived ...                                                   [OK]
    test_processEnded ...                                                  [OK]
    test_startError ...                                                    [OK]
    test_transport ...                                                     [OK]
    test_unicodeLogFileUTF8 ...                                            [OK]
    test_write ...                                                         [OK]
    test_writeSequence ...                                                 [OK]
  WorkerProtocolTests
    test_run ...                                                           [OK]
    test_start ...                                                         [OK]

-------------------------------------------------------------------------------
Ran 23 tests in 0.064s

PASSED (successes=23)
```

However, only 6 appear in test.log after the test run:

```
 grep -- '--> twisted.trial' _trial_temp/test.log 
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.LocalWorkerTests.test_transport <--
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.LocalWorkerTests.test_unicodeLogFileUTF8 <--
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.LocalWorkerTests.test_write <--
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.LocalWorkerTests.test_writeSequence <--
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.WorkerProtocolTests.test_run <--
2022-03-28 09:18:22-0400 [-] --> twisted.trial._dist.test.test_worker.WorkerProtocolTests.test_start <--
```

This is because the tests run application code that opens and truncates test.log.

<details><summary>Searchable metadata</summary>

```
trac-id__10321 10321
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__None None
time__1648473582503145 1648473582503145
changetime__1648475242422729 1648475242422729
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

https://github.com/twisted/twisted/pull/1710

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

Merged in bc42855619f6f909132d482f19b4797e031cd89e

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|
